### PR TITLE
View their entries

### PR DESCRIPTION
**Pull request by daisyyoo**
View Their Entries

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -18,10 +17,85 @@ function newEntry(event) {
...
+var $viewElements = document.querySelectorAll('.view');
+
+var $entriesNav = document.querySelector('a');
+var buttons = document.querySelectorAll('button');
+var $newEntryButton = buttons[1];
+$entriesNav.addEventListener('click', entryPage);
+$newEntryButton.addEventListener('click', newEntryPage);
+
+function entryPage(event) {
+  if (data.view === $viewElements[0].getAttribute('data-view')) {
```
Comment (codingcodymiller) [inline js/main.js]: What if you ever add a third view?

File: js/main.js [hunk 2]
```diff
@@ -18,10 +17,85 @@ function newEntry(event) {
...
+
+function refreshPage(event) {
+  if (data.view === $viewElements[1].getAttribute('data-view')) {
+    $viewElements[0].className = 'view hidden';
+    $viewElements[1].className = 'view ';
+  } else if (data.view === $viewElements[0].getAttribute('data-view')) {
+    $viewElements[0].className = 'view';
+    $viewElements[1].className = 'view hidden';
+  }
+}
```
Comment (codingcodymiller) [inline js/main.js]: Same problem here. This would be better off being a loop

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by daisyyoo** [comment]
View Their Entries Revised

File: js/main.js [hunk 3]
```diff
@@ -73,28 +73,44 @@ $entriesNav.addEventListener('click', entryPage);
...
-  if (data.view === $viewElements[1].getAttribute('data-view')) {
-    $viewElements[0].className = 'view hidden';
-    $viewElements[1].className = 'view ';
-  } else if (data.view === $viewElements[0].getAttribute('data-view')) {
-    $viewElements[0].className = 'view';
-    $viewElements[1].className = 'view hidden';
+  for (var i = 0; i < $viewElements.length; i++) {
+    if (data.view === $viewElements[i].getAttribute('data-view')) {
+      $viewElements[i].className = 'view hidden';
+      $viewElements[i].className = 'view ';
```
Comment (codingcodymiller) [inline js/main.js]: You only need the second one of these, the first one is getting immediately overwritten

File: js/main.js [hunk 4]
```diff
@@ -73,28 +73,44 @@ $entriesNav.addEventListener('click', entryPage);
...
+    } else if (data.view !== $viewElements[i].getAttribute('data-view')) {
+      $viewElements[i].className = 'view';
+      $viewElements[i].className = 'view hidden';
```
Comment (codingcodymiller) [inline js/main.js]: You only need the second one of these, the first one is getting immediately overwritten

File: js/main.js [hunk 5]
```diff
@@ -73,28 +73,44 @@ $entriesNav.addEventListener('click', entryPage);
...
-    $viewElements[1].className = 'view';
-    data.view = $viewElements[1].getAttribute('data-view');
+  for (var i = 0; i < $viewElements.length; i++) {
+    if (data.view === $viewElements[i].getAttribute('data-view')) {
+      $viewElements[i].className = 'view hidden';
+    } else {
+      $viewElements[i].className = 'view';
+    }
   }
+  currentView(event);
```
Comment (codingcodymiller) [inline js/main.js]: So this...

File: js/main.js [hunk 6]
```diff
@@ -73,28 +73,44 @@ $entriesNav.addEventListener('click', entryPage);
...
-    $viewElements[1].className = 'view hidden';
-    data.view = $viewElements[0].getAttribute('data-view');
+  for (var i = 0; i < $viewElements.length; i++) {
+    if (data.view === $viewElements[i].getAttribute('data-view')) {
+      $viewElements[i].className = 'view hidden';
+    } else {
+      $viewElements[i].className = 'view';
+    }
+  }
+  currentView(event);
```
Comment (codingcodymiller) [inline js/main.js]: Is exactly the same as this. You should make this a named function that you can call any time you want to do this.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**